internal/pubsub/rabbitpubsub/rabbit_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rabbitpubsub
+
+// To run these tests, first run:
+//     docker run -d --hostname my-rabbit --name rabbit -p 5672:5672 rabbitmq:3
+// Then wait a few seconds for the server to be ready.
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cloud/internal/pubsub/driver"
+	"github.com/streadway/amqp"
+)
+
+const rabbitURL = "amqp://guest:guest@localhost:5672/"
+
+func mustDialRabbit(t *testing.T) *amqp.Connection {
+	conn, err := amqp.Dial(rabbitURL)
+	if err != nil {
+		t.Skipf("skipping because the RabbitMQ server is not up (dial error: %v)", err)
```

Comment:
Have you figured out the record/replay strategy yet?

Reply:
It isn't feasible to do record/replay because the protocol is a custom one directly over TCP. 

I'm going with what Ross did on mysql: setup instructions to a human, along with skipping tests if the setup wasn't done.

Reply:
I guess this is fine for now, but I feel like we need to do better here. E.g., maybe we need to make our own fake.

https://github.com/NeowayLabs/wabbit ?

Reply:
Cool. Added issue #837.